pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -130,10 +165,20 @@ func redeployAgent(cluster *v3.Cluster, desiredAgent, desiredAuth string) bool {
 	}
 
 	if forceDeploy || imageChange || repoChange {
-		logrus.Infof("Redeploy Rancher Agents is needed: forceDeploy=%v, agent/auth image changed=%v,"+
-			" private repo changed=%v", forceDeploy, imageChange, repoChange)
+		logrus.Infof("Redeploy Rancher Agents is needed for %s: forceDeploy=%v, agent/auth image changed=%v,"+
+			" private repo changed=%v", cluster.Name, forceDeploy, imageChange, repoChange)
 		return true
 	}
+
+	na, ca := getAgentImages(cluster.Name)
+	if cluster.Status.AgentImage != na || cluster.Status.AgentImage != ca {
+		//downstream agent does not match, kick a redeploy with settings agent
+		clearAgentImages(cluster.Name)
```

Comment:
we already set the agent images in deploy logic, why do we need to clean it up here and L142? https://github.com/rancher/rancher/blob/b5409bab07aa3be1000ded896c05c74300a71956/pkg/controllers/management/clusterdeploy/clusterdeploy.go#L239

Reply:
just clear logic just in case, would prefer they were blank at the time we know they should be blank. youre right tho it's not technically necessary and would drop it if you insist

Reply:
gotcha, fine by me, just wanted to understand if there was some use case we were addressing with this